ts/screens/profile/EmailForwardingScreen.tsx

```diff
@@ -1,295 +1,183 @@
-        is_email_enabled: isEmailEnabled
-      })
-    );
-  }
-});
+  const setCustomEmailChannelEnabled = useCallback(
+    (customEmailChannelEnabled: boolean) => {
+      dispatch(customEmailChannelSetEnabled(customEmailChannelEnabled));
+    },
+    [dispatch]
+  );
+
+  const disableOrEnableAllEmailNotifications = useCallback(
+    (servicesId: ReadonlyArray<string>, profile: ProfileState) => {
+      const newBlockedChannels = getProfileChannelsforServicesList(
+        servicesId,
+        profile,
+        true,
+        "EMAIL"
+      );
+      dispatch(
+        profileUpsert.request({
+          blocked_inbox_or_channels: newBlockedChannels,
+          is_email_enabled: true
+        })
+      );
+    },
+    [dispatch]
+  );
 
-const EmailForwardingScreen = connect(
-  mapStateToProps,
-  mapDispatchToProps
-)(EmailForwardingScreenClass);
+  const bodyPropsArray: Array<BodyProps> = [
+    {
+      text: I18n.t("send_email_messages.subtitle")
+    },
+    {
+      text: <> {userEmail}</>,
+      weight: "Semibold"
+    },
+    {
+      text: I18n.t("global.symbols.question")
+    }
+  ];
+
+  const handleSwitchValueChange = useCallback(
+    (canSendEmail: boolean) => {
+      setIsLoading(true);
+      // eslint-disable-next-line functional/immutable-data
+      isCustomChannelEnabledChoice.current = false;
+      if (canSendEmail) {
+        disableOrEnableAllEmailNotifications(visibleServicesId, profile);
+      } else {
+        setEmailChannel(false);
+      }
+    },
+    [
+      disableOrEnableAllEmailNotifications,
+      profile,
+      setEmailChannel,
+      visibleServicesId
+    ]
+  );
 
-const EmailForwardingScreenFC = () => {
-  const navigation =
-    useNavigation<
-      IOStackNavigationProp<
-        ProfileParamsList,
-        "PROFILE_PREFERENCES_EMAIL_FORWARDING"
-      >
-    >();
+  useEffect(() => {
+    if (prevProfile && pot.isUpdating(prevProfile)) {
+      // if we got an error while updating the preference
+      // show a toast
+      if (pot.isError(profile)) {
+        IOToast.error(I18n.t("global.genericError"));
+        setIsLoading(false);
+        return;
+      }
+      // TODO move this login into a dedicated saga https://www.pivotaltracker.com/story/show/171600688
+      // if the profile updating is successfully then apply isCustomChannelEnabledChoice
+      if (
+        pot.isSome(profile) &&
+        isCustomChannelEnabledChoice.current !== undefined
+      ) {
+        setCustomEmailChannelEnabled(isCustomChannelEnabledChoice.current);
+        setIsLoading(false);
+        return;
+      }
+    }
+  }, [IOToast, profile, prevProfile, setCustomEmailChannelEnabled]);
 
-  return <EmailForwardingScreen navigation={navigation} />;
+  return (
+    <IOScrollViewWithLargeHeader
+      title={{ label: I18n.t("send_email_messages.title") }}
+      description={
+        (
+          <ComposedBodyFromArray body={bodyPropsArray} textAlign="left" />
+        ) as unknown as string
```

Comment:
What about extend `IOScrollViewWithLargeHeader`'s description prop to accept `string|JSX.Element`? This way, type casting can be avoided. What do you think?

Reply:
Actually, the description accepts string | undefined, so I needed the cast to avoid the TS error

Reply:
Actually, it is not very correct the forcing I have done. The correct solution would be to extend the `description` of  `IOScrollViewWithLargeHeader` so that it accepts a string or an array or undefined, to add [this logic](https://github.com/pagopa/io-app/blob/95377e52c69bbc3946ae2d3d6bfb4e048cdc6985/ts/components/screens/ScreenWithListItems.tsx#L67)  I had already thought about this modification but wanted to do it in an ad hoc PR.

Reply:
It makes sense!